app/not-a-bean/core/utils.ts

```diff
@@ -0,0 +1,106 @@
+import * as NotABean from '@/not-a-bean/models'
+
+export const NUM_SUITS = 4
+
+function generateDeck(): NotABean.Card[] {
+  const deck: NotABean.Card[] = []
+  for (let idx = 0; idx < 60; idx++) {
+    const cardSuit: number = Math.floor(idx / 15)
+    if (idx % 15 < 10) {
+      deck.push({
+        suit: cardSuit,
+        type: NotABean.CardType.NUMBER,
+        value: idx % 15,
+      })
+    } else {
+      if (idx % 15 === 10) {
+        deck.push({
+          suit: cardSuit,
+          type: NotABean.CardType.SPECIAL,
+          value: NotABean.SpecialCardValue.ZEROING,
+        })
+      } else if (idx % 15 === 11) {
+        deck.push({
+          suit: cardSuit,
+          type: NotABean.CardType.SPECIAL,
+          value: NotABean.SpecialCardValue.DOUBLE,
+        })
+      } else {
+        deck.push({
+          suit: cardSuit,
+          type: NotABean.CardType.SPECIAL,
+          value: NotABean.SpecialCardValue.NEGATION,
+        })
+      }
+    }
+  }
+  return deck
+}
+
+export function generateInitialGame(playerCount: number): NotABean.IGame {
+  if (60 % playerCount !== 0) {
+    throw new Error('Number of players should be divisor of 60')
+  }
+  const deck = generateDeck()
+  const game: NotABean.IGame = {
+    board: {
+      cardsPlayed: [],
+    },
+    firstPlayer: -1, // Initial game has first player ID -1, which is meaningless initial value
+    phase: {
+      payload: null,
+      type: NotABean.PhaseType.BeforeStart,
+    },
+    players: {},
+  }
+  for (let i = 0; i < playerCount; i++) {
+    game.players[i] = {
+      cardsAcquired: [],
+      hand: [],
+    }
+  }
+  // Deal the card
+  while (deck.length > 0) {
+    // Deal random card
+    const targetPlayer = deck.length % playerCount
+    const idx = Math.floor(Math.random() * deck.length)
+    game.players[targetPlayer].hand.push({
+      __meta: { knownTo: [targetPlayer] },
+      private: deck[idx],
+      public: {
+        isMouseOn: false,
+      },
+    })
+    // Remove card from the deck
+    deck.splice(idx, 1)
+  }
+  return game
+}
+
+export function calculateScore(cards: NotABean.Card[]): number {
+  let score: number = 0
+  for (let suit = 0; suit < NUM_SUITS; suit++) {
+    const filteredCards = cards.filter((card) => {
+      return card.suit === suit
+    })
+    score += filteredCards.reduce(addCardScore, 0) * filteredCards.reduce(aggSpecialCard, 1)
+  }
+  return score
+}
+
+function addCardScore(acc: number, card: NotABean.Card): number {
+  return acc + (card.type === NotABean.CardType.NUMBER ? card.value : 0)
+}
+
+function aggSpecialCard(multiplier: number, card: NotABean.Card): number {
+  if (card.type === NotABean.CardType.SPECIAL) {
+    if (card.value === NotABean.SpecialCardValue.DOUBLE) {
+      return multiplier * 2
+    } else if (card.value === NotABean.SpecialCardValue.NEGATION) {
+      return multiplier * (-1)
```

Comment:
위쪽 `return`문과 consistent하게 작성하려고 이렇게 쓰긴 했는데 굳이 그럴 필요는 없었으려나요

Reply:
스타일의 문제고 가독성 문제도 없으니, 사실 그대로 두어도 될 것 같습니다.

개인 선호의 영역인 듯도 한데, 이런 부분까지도 규칙을 정해서 우리 둘 다 그에 따를까요, 아니면 이런 부분은 개인의 선호로 둘까요? 코드 오너쉽에 대한 논의가 필요한 것 같기도 합니다.

좋은 아티클을 발견해서 공유드립니다. 의견 내 주시면 좋을 것 같습니다.

- https://docs.microsoft.com/en-us/azure/devops/learn/devops-at-microsoft/code-ownership-software-quality

Reply:
[Extreme Programming](https://xp123.com/xplor/xp0002e/index.shtml)에 따르면 다음과 같습니다.

> There are several approaches to a team's coding standard:
> *None*. Ugh.
> *Each chooses their own and uses it everywhere*. ("Then I can tell where I added code.") Ugh again.
> *Each chooses their own for their code, but uses existing style on existing code*. This is the minimum I can live with. Many times I've worked with a team that intended to have a standard, but evolved toward this as there was no "enforcement" pressure. At least with this, you can read an individual module. This approach does interfere with refactoring: when you move code between modules, now you have to reformat it as well.
> *Team has a style and sticks to it*. This is the ideal for XP. Since XP practices collective code ownership, it makes sense as well.

맨 마지막을 추구하는 것이 좋을 것 같습니다. 린터 옵션을 손볼 수 있다면 손보고,

Reply:
저 역시 마지막을 추구하는 쪽이 맞을것 같습니다. 근데 이런것도 린터 옵션으로 가능한가요? 뭔가 추상적인 내용이라 규칙을 정해야 할거같네요

Reply:
경우에 따라 옵션이 있는 경우도 있습니다.

https://palantir.github.io/tslint/rules/trailing-comma/

이 경우는 여기서 `enum`과 `typeLiterals`에 관한 부분을 지정하면 될 것 같습니다.

Reply:
린터 옵션 설정 관련한 이슈 #46 가 열렸으니 해당 코멘트 resolve 처리 하겠습니다